runtime/gc_glue_java/MarkingDelegate.hpp

```diff
@@ -276,7 +280,6 @@ class MM_MarkingDelegate
 typedef struct StackIteratorData4MarkingDelegate {
```

Comment:
let's try to move this up within MarkingDelegate
and use C++ style (no typedef), just 
```
struct StackIteratorData {
...
};
```

then later use it is `MM_MarkingDelegate::StackIteratorData`

Reply:
let's provide a basic constructor, too